oteapi_dlite/strategies/function.py

```diff
@@ -139,6 +135,12 @@ def get(
                 with open(f"{tmpdir}/data", "rb") as f:
                     cache.add(f.read(), key=key)
 
+        # __TODO__
+        # Can we savely assume that all strategies in a pipeline will be
+        # executed in the same Python interpreter?  If not, we should write
+        # the collection to a storage, such that it can be shared with the
```

Comment:
I think we cannot assume that, but maybe we need both options, as writing and reading is very slow compared to using the collection in memory

Reply:
I agree, avoiding unnecessary serialisations/deserialisations is important. We want to make a system that allows efficient handling of large datasets. In SFI PhysMet and MatCHMaker we will e.g. have SPED datasets that easily can be as large as 500 GB.

DLite is already highly optimised for sharing within one process, since all references to an instance with a given UUID refer to the same underlying memory. It is as efficient as numpy and like numpy it utilises reference counting.

When the evaluation of strategies are distributed over several processes or machines, we have to serialise/deserialise to/from a storage backend. This will be much less efficient, but probably needed with AiiDA. 

How to do this, is out of scope for this PR, but I added a [comment](https://github.com/EMMC-ASBL/oteapi-core/issues/176#issuecomment-1356562015) to  [issue 176](https://github.com/EMMC-ASBL/oteapi-core/issues/176) in oteapi-core.